### ISSUE TITLE
Training a loaded model with new nodes

### ISSUE THREAD

**Issue by ericbroda** [stellargraph/stellargraph#457]
How would the following scenario be done with StellarGraph - note that I am interested in step 3: Can the loaded model be trained with just the new nodes and not have to spend the time re-training the entire model from scratch?
1.  Train a large model (several tens of thousands of nodes)
2.  Save the model
3.  Identify new nodes, and have them added to the model WITHOUT retraining the entire model

**Comment by adocherty** [comment]
Hi @ericbroda,

Thanks for the question. Yes, some algorithms in StellarGraph can work in this way.  You will need to use an inductive algorithm which can transfer information between different graphs.

The best place to start is looking at the notebook that uses the GraphSAGE inductive algorithm to train on a subgraph of the Pubmed-Diabetes citation network, then add new nodes and predict on them using the pre-trained model.

From your issue I think that you also want to update the model by training on the new nodes. You can also do this with the GraphSAGE model, in the same way that you fine-tune any Keras model (for example this blog: https://flyyufelix.github.io/2016/10/03/fine-tuning-in-keras-part1.html)

See the demo here:
https://github.com/stellargraph/stellargraph/blob/develop/demos/node-classification/graphsage/graphsage-pubmed-inductive-node-classification-example.ipynb

Regards,
Andrew

**Comment by qillbel** [comment]
Hey @adocherty,

I have a similar question. How do I load a new graph with unlabeled nodes to do node classification with the trained model? 

I think I don't have to fine-tuned the my trained model because the labels are still the same in this task. (correct me if I'm wrong).

Thank you so much.

**Comment by adocherty** [comment]
Hi @qillbel,

Thanks for the question.  You're correct: you shouldn't have to fine-tune you're model (without labels that will be tricky anyway),  and you can simply load the new graph and create a new generator and use the `predict` method to perform node classification with the new graph.

For GraphSAGE we have a demo that does this here:
https://github.com/stellargraph/stellargraph/blob/develop/demos/node-classification/graphsage/graphsage-pubmed-inductive-node-classification-example.ipynb
